RestartManager.js

```diff
@@ -3,41 +3,54 @@ const NativeCodePush = require("react-native").NativeModules.CodePush;
 
 const RestartManager = (() => {
     let _allowed = true;
-    let _restartPending = false;
+    let _restartInProgress = false;
+    let _restartQueue = [];
```

Comment:
When you say "squash" them, I assume you mean drop the second one if the first one is in progress.

The problem is with `onlyIfUpdateIsPending`. Lets say we have this scenario:

- The first time, `onlyIfUpdateIsPending` is `true`.
- The second time, `onlyIfUpdateIsPending` is `false`. 
- There is no update pending.

If we allowed them to run in parallel, the second one would succeed. But if we dropped the second one because the first one is in progress, the customer would see the restart not actually happen despite setting `onlyIfUpdateIsPending` to `false`, which is not the right behaviour based on our documentation.